### PR TITLE
Add tests for shift operators in DependencyMixin

### PR DESCRIPTION
**Pull request by raphaelauv**
add missing tests on taskmin before fixing https://github.com/apache/airflow/discussions/36088

### CODE REVIEW

File: tests/models/test_taskmin.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            op_b = EmptyOperator(task_id="b")
+            op_c = EmptyOperator(task_id="c")
+            op_d = EmptyOperator(task_id="d")
+
+            op_a >> [op_b >> op_c, op_d]
+
+        dag_maker.create_dagrun()
+        assert op_a in op_c.upstream_list
+        assert [op_a] == op_d.upstream_list
+        assert [] == op_b.upstream_list
```
Comment (dirrao) [inline tests/models/test_taskmin.py]: why op_b upstream is empty. isn't it should  be op_a?

Comment (raphaelauv) [reply tests/models/test_taskmin.py]: This is the current behavior, check the discussion I have link in the description of this PR

File: tests/models/test_taskmin.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import pytest
+
+from airflow.operators.empty import EmptyOperator
+
+pytestmark = pytest.mark.db_test
```
Comment (hussein-awala) [inline tests/models/test_taskmin.py]: I wonder why do we need db in these tests 🤔

File: tests/models/test_taskmin.py (hunk 1 above)
Comment (hussein-awala) [reply tests/models/test_taskmin.py]: yes, this is the behavior of Python shift operators, I hope users don't define task dependencies this way

File: tests/models/test_taskmin.py (hunk 2 above)
Comment (raphaelauv) [reply tests/models/test_taskmin.py]: done

Comment (hussein-awala) [reply tests/models/test_taskmin.py]: looks like these two tests need the DB access

**Comment by raphaelauv** [comment]
@hussein-awala could you please re-review pls

**Review by hussein-awala — APPROVED** [review]

**Review by hussein-awala — CHANGES_REQUESTED** [review]
After rechecking it, I think that the tests should be in https://github.com/apache/airflow/blob/1aa91a482630f389005a9a36b0f038e675163411/tests/models/test_taskmixin.py, no? and I think the class name should be TestDepencyMixin.

If so, please move the class and update its name, then update the PR title to something more meaningful and clearer.

**Comment by raphaelauv** [comment]
@hussein-awala thanks , it's done

**Review by hussein-awala — APPROVED** [review]

**Pull request merged**